### PR TITLE
ENH: Require libtiff >=4.0.3

### PR DESCRIPTION
**Pull request by blowekamp**
Remove macros and try compiles to support old libtif versions.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]
Is this PR fully independent from #4822?

**Comment by dzenanz** [comment]
With the other PR merged, maybe rebase this one? Also, remove draft marking?

**Comment by blowekamp** [comment]
Yes, thank you.

File: Modules/ThirdParty/TIFF/CMakeLists.txt [hunk 1]
```diff
@@ -8,29 +8,20 @@ if(ITK_USE_SYSTEM_TIFF)
...
     )
   set(ITKTIFF_LIBRARIES "${TIFF_LIBRARIES}")
   set(ITKTIFF_NO_SRC 1)
 
-  # Check availability of different methods to access TIFF's field
-  # data structure.
-  include(CheckTypeSize)
-  set(CMAKE_REQUIRED_INCLUDES ${ITKTIFF_SYSTEM_INCLUDE_DIRS})
-  set(CMAKE_EXTRA_INCLUDE_FILES "tiffio.h")
-  check_type_size(TIFFFieldReadCount ITK_TIFF_HAS_TIFFFieldReadCount)
```
Comment (blowekamp) [inline Modules/ThirdParty/TIFF/CMakeLists.txt:23]: These are in a ".h.in" file with a `#cmakedefine`. That file appears to be missing from this PR.

**Comment by dzenanz** [comment]
```log
Error when uploading file: /Users/runner/work/1/s-build/Testing/20240829-1318/Update.xml
   Error message was: Recv failure: Operation timed out
```
This is probably due to some network infrastructure maintenance at Kitware this morning.

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
Thanks, Brad!

**Pull request merged**